### PR TITLE
_msearch subrequests parsing #4227

### PR DESCRIPTION
**Pull request by vieirajlt**
`MultiSearchRequest.add()` relies on `RestSearchAction.parseSearchRequest()` to parse the subrequests by making parsed vars on each subrequest part of `RestRequest.params` .

Closes #4227

### CODE REVIEW

**Comment by mayya-sharipova** [comment]
@elasticmachine test this please

File: server/src/main/java/org/elasticsearch/action/search/MultiSearchRequest.java [hunk 1]
```diff
@@ -78,6 +78,13 @@ public MultiSearchRequest add(SearchRequest request) {
         return this;
     }
 
+    public MultiSearchRequest add(SearchRequest searchRequest, RestRequest request, XContentParser parser) throws IOException {
+        IntConsumer setSize = size -> searchRequest.source().size(size);
+        RestSearchAction.parseSearchRequest(searchRequest, request, parser, setSize, false);
```
Comment (javanna) [inline server/src/main/java/org/elasticsearch/action/search/MultiSearchRequest.java]: Ideally we would not reference any Rest action from a request

File: server/src/main/java/org/elasticsearch/rest/RestRequest.java [hunk 2]
```diff
@@ -270,6 +271,14 @@ public final String param(String key, String defaultValue) {
         return params;
     }
 
+    public void params(Map<String, String> params) {
+        for (Map.Entry<String, String> entry : params.entrySet()) {
+            if(this.params.get(entry.getKey()) == null) {
+                this.params.put(entry.getKey(), entry.getValue());
+            }
+        }
+    }
```
Comment (javanna) [inline server/src/main/java/org/elasticsearch/rest/RestRequest.java]: I was hoping that we could have shared code between search and msearch to keep the supported params in sync, without having to change RestRequest. Do you think that is possible?

File: server/src/test/java/org/elasticsearch/action/search/MultiSearchRequestTests.java [hunk 3]
```diff
@@ -178,7 +185,7 @@ public void testSimpleAdd4() throws Exception {
         assertThat(request.requests().get(2).types()[0], equalTo("type2"));
         assertThat(request.requests().get(2).types()[1], equalTo("type1"));
         assertThat(request.requests().get(2).routing(), equalTo("123"));
-    }
+    }*/
```
Comment (javanna) [inline server/src/test/java/org/elasticsearch/action/search/MultiSearchRequestTests.java]: should the commented out tests be moved somewhere else?

**Review by javanna — CHANGES_REQUESTED** [review]
hi @vieirajlt thanks for your PR, I left a couple of comments. Did you have a chance to check which parameters are missing in msearch that search supports, just to see what we are adding support for with your change ?

**Comment by jimczi** [comment]
@vieirajlt sorry for the late reply, are you still interested in getting this PR in? It has merge conflicts that need to be addressed and @javanna left some comments regarding the implementation.

**Comment by javanna** [comment]
Closing for lack of feedback.

**Pull request closed without merging**